pkg/avfoundation/avfoundation_darwin.go

```diff
@@ -101,6 +101,7 @@ type ReadCloser struct {
 	cancelCtx  context.Context
 	cancelFunc func()
 	closeWG    *sync.WaitGroup
+	lock       *sync.Mutex
```

Comment:
```suggestion
	lock       sync.Mutex
```
then we don't need to initialize like `rc.lock = &sync.Mutex{}`

(above WaitGroup as well)